### PR TITLE
build: pin versions of NPM global deps in `Makefile`

### PR DESCRIPTION
**Pull request by agilgur5**
### Motivation

Make sure that there are no version inconsistencies due to NPM global deps installed in the `Makefile` (which is used locally as well as in CI)
- This has been a bit concerning to me ever since I noticed the installs are not pinned 😅 
- This also makes the "# update this in Nix" comments make more sense as there's a version specified now

### Modifications

- add `@<version>` for each `npm i -g`
  - use the versions from [`dev/nix/node-packages.nix`](https://github.com/argoproj/argo-workflows/blob/7f22085d16316bdd77bdb1060455140d9d6e1664/dev/nix/node-packages.nix#L853) and [`dev/nix/package.json#dependencies`](https://github.com/argoproj/argo-workflows/blob/7f22085d16316bdd77bdb1060455140d9d6e1664/dev/nix/package.json#L5)

### Verification

Ran in my devcontainer successfully:

```bash
npm i -g markdown-spellcheck@1.3.1
npm i -g markdownlint-cli@0.33.0
```

### CODE REVIEW

File: Makefile
Comment (terrytangyuan) [inline Makefile:647]: Can you also up this in Nix (dev folder)?

Comment (agilgur5) [reply Makefile:647]: Per the PR description, this _is_ the version used in the Nixfiles

Comment (agilgur5) [reply Makefile:647]: Unless you meant updating all of the above to [latest](https://github.com/igorshubovych/markdownlint-cli/releases/tag/v0.35.0)?

**Review by terrytangyuan — APPROVED** [review]

**Comment by terrytangyuan** [comment]
Thanks!

File: Makefile
Comment (agilgur5) [reply Makefile:647]: Resolved discussion as this PR was approved, so I am assuming this good to go now. Just waiting on merge

**Pull request merged**
